### DISCUSSION TITLE
Which Minuit2 version/patch_level/commit is used in iminuit?

### DISCUSSION THREAD

**Discussion by JacekHoleczek** [scikit-hep/iminuit#948] [Q&A]
As of today (2023.11.06), the last [*iminuit*](https://github.com/scikit-hep/iminuit) release is `v2.24.0` (2023.08.15).
I cannot find any place that says which [*Minuit2*](https://github.com/root-project/root/tree/master/math/minuit2) version/patch_level/commit it incorporates.
Well, the last [*Minuit2*](https://github.com/root-project/root/tree/master/math/minuit2) improvements/fixes are just two weeks old (2023.10.25).

**Comment by HDembinski** [comment]
The ROOT version against which iminuit is built is shown on the frontpage of the docs. iminuit is occasionally updated to the latest ROOT version. We don't do this regularly, because most of the changes in Minuit2 are not relevant to us.
We have our own extensive suite of unit tests to ensure that iminuit works correctly.

Unless you have a specific fix in mind that you want to see in iminuit, I think you shouldn't worry about this.

**Reply by HDembinski** [reply]
In fact, often the "improvements" in ROOT are adding new bugs, which then creates work for me to write workarounds in iminuit. This is why I don't update the C++ code regularly.

To demonstrate: I just updated ROOT to current main and now a test which was passing before fails and it is a bug in C++ that was introduced by the ROOT devs.

**Reply by JacekHoleczek** [reply]
It seems to me that it would really be a good idea to keep the [*Minuit2*](https://github.com/root-project/root/tree/master/math/minuit2) version up-to-date (reporting any found problems/errors back to [*ROOT*](https://github.com/root-project/root/issues)).

I've seen a "report" (during a non-public collaboration meeting) that "SIMPLEX" in [*iminuit*](https://github.com/scikit-hep/iminuit) gave "bad results" (when compared to the same method in [ROOT](https://root.cern.ch/)). I did not try to follow this "report", nor I tried to play with it (I don't think it would be easy to create a "standalone reproducer").

**Reply by HDembinski** [reply]
That's exactly what is happening, I already fixed several bugs in Minuit2 in ROOT, you can check my PRs in the ROOT repo.

That being said, it is not in my interest to create more work for myself and therefore iminuit won't be regularly updated to the latest ROOT version. In recent years there have been no fundamental changes to the core Minuit2 code that made it worthwhile to update frequently.
